Base/QTCore/qSlicerExtensionsManagerModel.cxx

```diff
@@ -1693,7 +1693,7 @@ bool qSlicerExtensionsManagerModel::downloadAndInstallExtension(const QString& e
   // Ensure extension metadata has been downloaded from the server
   if (d->ExtensionsMetadataFromServer.empty())
     {
-    this->updateExtensionsMetadataFromServer(false, true);
+    this->updateExtensionsMetadataFromServer(/* force= */true, /* waitForCompletion= */true);
```

Comment:
Does "force" trigger a new download of all metadata of all extensions? That would add unnecessary delay and server load every time an extension is installed. We should only query the server if no metadata is available.

When the extension manager is opened, all extensions metadata are downloaded automatically. Have you run into this issue when using the Slicer GUI or using the extension manager model API from code?

Reply:
> Does "force" trigger a new download of all metadata of all extensions? 

It does not because `updateExtensionsMetadataFromServer` is only called from `downloadAndInstallExtension` when `ExtensionsMetadataFromServer` is empty. 

> When the extension manager is opened, all extensions metadata are downloaded automatically.

This would be the case with `AutoCheck` is enabled, I was testing in the case of a fresh install of Slicer without any settings set. 

In that case, the download of metadata need to happen once when the user click the first time on the "Install" button (associated with `downloadAndInstallExtension`).

Reply:
> Have you run into this issue when using the Slicer GUI

This happened when using the GUI

Reply:
Ok, it's all good then.

I wanted to implement downloading of all extensions metadata when the user opens the extensions manager, but apparently I did not get to it.